src/components/contextual/stake/StakePreview.vue

```diff
@@ -51,7 +51,12 @@ const {
 } = useStaking();
 const { getTokenApprovalActionsForSpender } = useTokenApprovalActions(
   [props.pool.address],
-  ref([balanceFor(props.pool.address).toString()])
+  ref([
+    (props.action === 'restake'
+      ? stakedSharesForProvidedPool.value
+      : balanceFor(props.pool.address)
+    ).toString(),
+  ])
```

Comment:
Not a massive fan of having conditional logic within function calls like this. Could we extract the ref and call it something readable?

Reply:
Updated